### PR TITLE
Add account creation feature

### PR DESCRIPTION
**Pull request by mdorrance**
Initial commit for WIP

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -38,5 +38,15 @@
 
   add_index "products", ["category_id"], name: "index_products_on_category_id", using: :btree
 
+  create_table "users", force: :cascade do |t|
```
Comment (sabondano) [inline db/schema.rb]: Don't we need a username in the users table? The user story talks about a username. Do we want to use email when logging in instead?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -10,4 +10,10 @@ def load_featured_products
   def cart
     @cart ||= Cart.new(session[:cart])
   end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```
Comment (sabondano) [inline app/controllers/application_controller.rb]: Should we add an instance variable assignment with memoization on this line for the server not to have to run this query every single time? What do you all think?

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+  let(:user) do
+    User.new(first_name: "Jane",
+             last_name:  "Doe",
+             email:      "jane@doe.com",
+             password:   "password")
```
Comment (mdorrance) [inline spec/models/user_spec.rb]: Team agrees to ignore.

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+  let(:user) do
+    User.new(first_name: "Jane",
+             last_name:  "Doe",
+             email:      "jane@doe.com",
```
Comment (mdorrance) [inline spec/models/user_spec.rb]: Team agrees to ignore.

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+  let(:user) do
+    User.new(first_name: "Jane",
+             last_name:  "Doe",
```
Comment (mdorrance) [inline spec/models/user_spec.rb]: Team agrees to ignore.

File: app/controllers/sessions_controller.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
```
Comment (mdorrance) [inline app/controllers/sessions_controller.rb]: Team agrees to ignore.

File: db/schema.rb (hunk 1 above)
Comment (mdorrance) [inline db/schema.rb]: Team agrees to use Email for Username.

**Pull request merged**